### ISSUE TITLE
Gitlabform 3.X not adding new member/approver correctly.

### ISSUE THREAD

**Issue by codrcodz** [gitlabform/gitlabform#461]
**Describe the bug**

Observed Behavior:
*Given* a repository with an existing set of users and group and an existing set of approvers and approver groups,
*When* I add a new user and simultaneously add that same user as an approver,
*Then* gitlabform will add the user as a member, but not an approver. 

Expected Behavior:
*Given* a repository with an existing set of users and group and an existing set of approvers and approver groups,
*When* I add a new user and simultaneously add that same user as an approver,
*Then* gitlabform will add the user as a member AND as an approver.

_When the same configs are run a second time, the member is then successfully added to the approver list. Gitlabform returns zero on the first run, when it fails to add the new member as an approver. It successfully adds the (now existing) member on the second run. Example configuration:_

Initial config (before adding new user):

```
config_version: 3
gitlab:
  url: "https://gitlab.myorg.org"
  ssl_verify: true
  timeout: 10
  "myorg/*":
    group_settings:
      description: "Repositories managed by myorg."
      visibility: internal
      project_creation_level: maintainer
      subgroup_creation_level: maintainer
      request_access_enabled: false
      emails_disabled: true
      lfs_enabled: true
      default_branch_protection: 2
    group_members:
      users:
        gitlabform-svc-acct:
          access_level: 50
      enforce: true
    merge_requests:
      approvals:
        approvals_before_merge: 1
        reset_approvals_on_push: true
        disable_overriding_approvers_per_merge_request: true
      approver_groups:
        - myorg
      remove_other_approval_rules: true
    project_settings:
      visibility: internal
      only_allow_merge_if_pipeline_succeeds: true
      only_allow_merge_if_all_discussions_are_resolved: true
      remove_source_branch_after_merge: true
      squash_option: default_off
      wiki_access_level: disabled
      keep_latest_artifact: true
    branches:
      master:
        protected: true
        push_access_level: 0
        merge_access_level: 40
        unprotect_access_level: 40
      main:
        protected: true
        push_access_level: 0
        merge_access_level: 40
        unprotect_access_level: 40
"myorg/mysuborg/*":
    group_settings:
      visibility: internal
      project_creation_level: maintainer
      subgroup_creation_level: maintainer
      request_access_enabled: false
      emails_disabled: true
      lfs_enabled: true
      default_branch_protection: 2
    group_members:
      users:
        anotheruser:
          access_level: 40
      enforce: true
myorg/mysuborg/myproject:
    members:
      users:
        codrcodz:
          access_level: 40
      enforce: true
    merge_requests:
      approvals:
        approvals_before_merge: 1
        reset_approvals_on_push: true
        disable_overriding_approvers_per_merge_request: true
      approvers:
        - codrcodz
      remove_other_approval_rules: true
```

New config (adding the new user: "newuser"):
```
config_version: 3
gitlab:
  url: "https://gitlab.myorg.org"
  ssl_verify: true
  timeout: 10
  "myorg/*":
    group_settings:
      description: "Repositories managed by myorg."
      visibility: internal
      project_creation_level: maintainer
      subgroup_creation_level: maintainer
      request_access_enabled: false
      emails_disabled: true
      lfs_enabled: true
      default_branch_protection: 2
    group_members:
      users:
        gitlabform-svc-acct:
          access_level: 50
      enforce: true
    merge_requests:
      approvals:
        approvals_before_merge: 1
        reset_approvals_on_push: true
        disable_overriding_approvers_per_merge_request: true
      approver_groups:
        - myorg
      remove_other_approval_rules: true
    project_settings:
      visibility: internal
      only_allow_merge_if_pipeline_succeeds: true
      only_allow_merge_if_all_discussions_are_resolved: true
      remove_source_branch_after_merge: true
      squash_option: default_off
      wiki_access_level: disabled
      keep_latest_artifact: true
    branches:
      master:
        protected: true
        push_access_level: 0
        merge_access_level: 40
        unprotect_access_level: 40
      main:
        protected: true
        push_access_level: 0
        merge_access_level: 40
        unprotect_access_level: 40
"myorg/mysuborg/*":
    group_settings:
      visibility: internal
      project_creation_level: maintainer
      subgroup_creation_level: maintainer
      request_access_enabled: false
      emails_disabled: true
      lfs_enabled: true
      default_branch_protection: 2
    group_members:
      users:
        anotheruser:
          access_level: 40
      enforce: true
myorg/mysuborg/myproject:
    members:
      users:
        codrcodz:
          access_level: 40
        newuser:
          access_level: 40
      enforce: true
    merge_requests:
      approvals:
        approvals_before_merge: 1
        reset_approvals_on_push: true
        disable_overriding_approvers_per_merge_request: true
      approvers:
        - codrcodz
        - newuser
      remove_other_approval_rules: true
```

Notes:

`gitlabform` is run against "ALL", not an explicit subset of repos or groups, like so:

`gitlabform -c config.yml ALL`

In the above configs, I have sanitized my organization's information and usernames from the configs,
and stripped out all the projects and suborgs but one, but all the other settings are as you see them. This behavior is happening on multiple projects, below several different suborgs, all under the same top-level org with the settings you see under `myorg/*` in the above configuration.

**GitLabForm version**
🏗  GitLabForm version: 3.2.0 = outdated 😔, please update! (the latest stable is 3.3.2)

**GitLab version**
GitLab Enterprise Edition [15.3.2-ee](https://gitlab.com/gitlab-org/gitlab/-/tags/v15.3.2-ee)

**Comment by gdubicki** [comment]
Sorry for not getting back to you earlier. This should be fixed in [v3.3.3](https://github.com/gitlabform/gitlabform/releases/tag/v3.3.3) stable and all future versions.

**Comment by codrcodz** [comment]
Thanks a lot! I'll make sure to upgrade our wrapper utility that we call gitlabform with to use this newer version as soon as possible.
